src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs

```diff
@@ -158,6 +160,43 @@ public static object MapSprint(string iterationPathsString)
             return iterationPath;
         }
 
+        private static readonly Dictionary<string, decimal> CalculatedLexoRanks = new Dictionary<string, decimal>();
+        private static readonly Dictionary<decimal, string> CalculatedRanks = new Dictionary<decimal, string>();
+
+        private static readonly Regex LexoRankRegex = new Regex(@"^[0-2]\|[0-9a-zA-Z]*(\:[0-9a-zA-Z]*)?$",
+            RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.Singleline);
+
+        public static object MapLexoRank(string lexoRank)
+        {
+            if (string.IsNullOrEmpty(lexoRank) || !LexoRankRegex.IsMatch(lexoRank))
+                return decimal.MaxValue;
+
+            if (CalculatedLexoRanks.ContainsKey(lexoRank))
+            {
+                Logger.Log(LogLevel.Warning, "Duplicate rank detected. You may need to re-balance the JIRA LexoRank. see: https://confluence.atlassian.com/adminjiraserver/managing-lexorank-938847803.html");
```

Comment:
I think exceptions that actually tell me what to do to fix them are fantastic :)

Reply:
Always do this as I kept loosing the link while trying to get this working. Lol